src/main/java/com/the/hugging/team/utils/Dialogs.java

```diff
@@ -242,6 +246,271 @@ else if (dds.getText().length() != 11 &&
+            @Override
+            public String toString(ProductCategory productCategory) {
+                return productCategory.getName();
+            }
+
+            @Override
+            public ProductCategory fromString(String s) {
+                return null;
+            }
+        };
+        category.setConverter(categoryConverter);
+        category.getSelectionModel().selectFirst();
+
+        ChoiceBox<ProductQuantityType> quantityType = new ChoiceBox<>(FXCollections.observableArrayList(productQuantityTypeService.getAllProductQuantityTypes()));
+        StringConverter<ProductQuantityType> productQuantityTypeConverter = new StringConverter<>() {
+            @Override
+            public String toString(ProductQuantityType productQuantityType) {
+                return productQuantityType.getName();
+            }
+
+            @Override
+            public ProductQuantityType fromString(String s) {
+                return null;
+            }
+        };
+        quantityType.setConverter(productQuantityTypeConverter);
+        quantityType.getSelectionModel().selectFirst();
+
+        ChoiceBox<Shelf> shelf = new ChoiceBox<>(FXCollections.observableArrayList(storageService.getAllShelves()));
+        StringConverter<Shelf> shelfStringConverter = new StringConverter<>() {
+            @Override
+            public String toString(Shelf shelf) {
+                return shelf.getName();
+            }
+
+            @Override
+            public Shelf fromString(String s) {
+                return null;
+            }
+        };
+        shelf.setConverter(shelfStringConverter);
+        shelf.getSelectionModel().selectFirst();
+
+        if (product.getName() != null) {
+            productName.setText(product.getName());
+        }
+
+        if (product.getNomenclature() != null) {
+            nomenclature.setText(product.getNomenclature());
+        }
+
+        if (product.getProductCategory() != null) {
+            category.getSelectionModel().select(product.getProductCategory());
+        }
+
+        if (product.getQuantity() != null) {
+            quantityAmount.setText(product.getQuantity().toString());
+        }
+
+        if (product.getRetailPrice() != null) {
+            retailPrice.setText(product.getRetailPrice().toString());
+        }
+
+        if (product.getWholesalePrice() != null) {
+            wholesalePrice.setText(product.getWholesalePrice().toString());
+        }
+
+        if (product.getDeliveryPrice() != null) {
+            deliveryPrice.setText(product.getDeliveryPrice().toString());
+        }
+
+        if (product.getProductQuantityType() != null)
+        {
+            quantityType.getSelectionModel().select(product.getProductQuantityType());
+        }
+
+        if (product.getShelf() != null)
+        {
+            shelf.getSelectionModel().select(product.getShelf());
+        }
+
+        TextField newQuantityTypeName = new TextField();
```

Comment:
I don't think we need to have creating new quantity types and product categories because they are static data. If we need to add a new type or category we will add it through the DB.